frontend/src/app/containers/ExperimentPage/index.js

```diff
@@ -65,37 +64,47 @@ export class ExperimentDetail extends React.Component {
 
     const {
       isExperimentEnabled,
-      experiment,
-      getCookie,
-      hasAddon
+      experiment
     } = this.props;
 
-    let showEmailDialog = false;
-    if (getCookie("visit-count") === "2") {
-      if (hasAddon) {
-        showEmailDialog = true;
-      }
-    }
-
     // TODO: Clean this up per #1367
     this.state = {
       enabled: isExperimentEnabled(experiment),
       highlightMeasurementPanel: false,
       isEnabling: false,
       isDisabling: false,
       progressButtonWidth: null,
-      showEmailDialog,
+      showEmailDialog: false,
       showDisableDialog: false,
-      shouldShowTourDialog: false,
       showTourDialog: false,
       showPreFeedbackDialog: false,
       showEolDialog: false
     };
   }
 
-  componentWillReceiveProps(nextProps: ExperimentDetailProps) {
-    const { shouldShowTourDialog, enabled: prevEnabled } = this.state;
+  checkCookies() {
+    const { getCookie, removeCookie, experiment } = this.props;
+    const exp = getCookie("exp-installed");
+    if (getCookie("txp-installed")) {
+      removeCookie("txp-installed");
+      this.setState({ showEmailDialog: true });
+    } else if (exp && !this.state.showEmailDialog) {
+      if (experiment && experiment.addon_id === exp) {
+        removeCookie("exp-installed");
+        this.setState({ showTourDialog: true });
+      }
+    }
```

Comment:
Installing the testpilot addon sets txp-installed and experiment install sets exp-installed. When both are set with a one-click, this shows the email dialog first, then the tour. Since they're cookies they can persist through a page refresh.

I don't want to continue using cookies here, I think we need a more organized general way to persist state, but I think that should come in later.

Reply:
I'd like to see us add like a PopupManager container that'd manage what popups to show.